samples/nrf_desktop/src/hw_interface/battery_discrete.c

```diff
@@ -420,32 +453,11 @@ static bool event_handler(const struct event_header *eh)
 	}
 
 	if (is_power_down_event(eh)) {
-		if (atomic_get(&active)) {
-			atomic_set(&active, false);
-
-			int err = gpio_pin_disable_callback(gpio_dev,
-					CONFIG_DESKTOP_BATTERY_CSO_PIN);
-
-			if (!err) {
-				err = cso_pin_control(false);
-			}
-
-			if (err) {
-				module_set_state(MODULE_STATE_ERROR);
-			}
-
-			if (adc_async_read_pending) {
-				__ASSERT_NO_MSG(sampling);
-				/* Poll ADC and postpone shutdown */
-				k_delayed_work_submit(&battery_lvl_read,
-						      K_MSEC(0));
-			} else {
-				/* No ADC sample left to read, go to standby */
-				battery_monitor_stop();
-			}
+		if (battery_state != BATTERY_STATE_ERROR) {
```

Comment:
there should be no need for that - error would mean no sampling and sleep should do no action as module is not active.